### PR TITLE
refactor(tokens): flatten ds4/ds6 folders

### PR DESCRIPTION
**Pull request by ianmcburnie**
Issue #1646

- [ ] This PR contains CSS changes
- [X] This PR does not contain CSS changes

## Description

EDIT: This was _supposed_ to be a DRAFT PR request showing the proposed direction/step, but I clicked the wrong button. So just imagine that it is...

* Removes ds4/ds6 folders under each module (only done badge, switch & tabs so far - which should give the idea - before I proceed any further with the rest)
* Uses a global less variable (passed in via CLI) configuration, instead of file/folder convention, to dictate ds4 or ds6 
* Leaves src file under `/base` subfolder, but we would probably want to flatten that folder too (I didn't here to keep diff cleaner)

FYI this is not the final end state! Just another step on the way. This PR gets us one hop closer to either:

a) maintaining ds4 dist folders and using only a LESS tokens file to dictate differences between ds4 & ds6 (which currently all sit in the `primitive.less`)
b) removing ds4 dist folders entirely and using a custom property file to dictate ds4 overrides (i.e. exposing all core tokens as props, and over-riding them in `legacy-theme.less`)

The key thing that is happening here is that we have switched to a configuration over convention style, i.e. we are passing in a global less variable which dictates which bundle of LESS tokens is included in the compilation - which will give us either a ds4 or ds6 output. Prior to this PR, we depended on a file/folder convention (we needed this as in the past there were many differences between ds4 and ds6 that could not be expressed through a token system and global var alone). We can further extend this approach to output things other than ds4 or ds6, given any input of LESS tokens , e.g. a "seasonal" theme, even via an external package dependency.

Currently I am leaning towards A, as I'm not sure it's a good idea to expose every core token as a custom property.

Whichever way we ultimately go before v14 release, this should be a useful interim step as it flattens the ds4/ds6 folders under each module and also restructures the gulpfile a little for some more fine grained control over how/when things are compiled by lessc (we can now compile individual modules via CLI - which we can use to our advantage to speed up browser refresh/sync when developing locally)

## Notes

This branch will not build css locally because I haven't updated all modules yet to consume the global LESS variable. Only a few.

## Checklist

- [ ] I verify the build is in a non-broken state (see notes above)
- [X] I verify all changes are within scope of the linked issue

### CODE REVIEW

**Review by ArtBlue — COMMENTED** [review]
Will the script(s) running the gulpfile through `package.json`  be revised to append the ds6 param automatically? Or will ds6 be assumed and defaulted somewhere?

**Comment by ArtBlue** [comment]
> 

Or maybe if it's going to be a single thread anyway at some point, that's not even necessary?

**Comment by ianmcburnie** [comment]
Yeah, I imagine the package scripts might get updated a little. Although it's also easy to just call these scripts directly, e.g. `gulp compileModule --name switch --ds ds6`

File: gulpfile.js [hunk 1]
```diff
@@ -127,7 +144,10 @@ function syncDocsHtml(cb) {
         .on('close', cb);
 }
 
-// public tasks listed below
-
+// public tasks
+exports.compileModule = compileModule;
+exports.compileModules = compileModules;
+exports.compileBundle = compileBundle;
+exports.compileBundles = compileBundles;
```
Comment (agliga) [inline gulpfile.js:151]: Why are these duplicated?

Comment (ianmcburnie) [reply gulpfile.js:151]: They are all different. Can now compile an individual module or bundle, or compile all modules or bundles.

Comment (agliga) [reply gulpfile.js:151]: Got it. Just seemed a bit confusing but i see the plural now.

Comment (agliga) [reply gulpfile.js:151]: Can we maybe do
compileModule
compileAllModules?
That way its more intuitive than just a plural

Comment (ianmcburnie) [reply gulpfile.js:151]: Yeah. Good idea.

**Pull request closed without merging**